### PR TITLE
feat(curriculum): Added Bash and SQL Quiz

### LINKED ISSUE #56233: [HACKTOBERFEST 2024] - Bash and SQL Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Bash and SQL quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql`.

**All quiz questions need to relate to one of these topics:**

- What is SQL
- SQL data types
- basic commands and queries
- primary and foreign keys
- many-to-many relationships
- normalisation
- SQL injection and preventing it

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by AbhinavTheDev**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56233

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 1]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 
 #### --text--
 
-Placeholder question
+What does SQL stands for?
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:20]:
```suggestion
What does SQL stand for?
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 2]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 2
+They are archived.
 
 ---
 
-Placeholder distractor 3
+They are ophaned.
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:318]:
```suggestion
They are orphaned.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Review by moT01 — CHANGES_REQUESTED** [review]
All of the code (or fake code) should be placed in backticks @AbhinavTheDev. For example, `GENERATE DATABASE`, `CREATE DATABASE`, `SELECT []`, etc. should be `` `GENERATE DATABASE` ``, `` `CREATE DATABASE` ``, `` `SELECT []` ``. along with all the others.

**Review by zairahira — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 3]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`SELECT *`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does FOREIGN KEY establish?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:108]:
```suggestion
What does `FOREIGN KEY` establish?
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 4]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --distractors--
 
-Placeholder distractor 1
+A unique identifier key
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:112]:
```suggestion
A unique identifier key.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 5]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 2
+A primary key
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:116]:
```suggestion
A primary key.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 6]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 3
+A null constraint
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:120]:
```suggestion
A null constraint.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 7]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --text--
 
-Placeholder question
+In a junction table, what do you typically include?
 
 #### --distractors--
 
-Placeholder distractor 1
+Only foreign keys
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:200]:
```suggestion
Only foreign keys.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 8]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 2
+Only timestamps
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:204]:
```suggestion
Only timestamps.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 9]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 3
+Only primary keys
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:208]:
```suggestion
Only primary keys.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 10]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --answer--
 
-Placeholder answer
+Foreign keys and additional data
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:212]:
```suggestion
Foreign keys and additional data.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 11]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --text--
 
-Placeholder question
+Which key can have duplicate values?
 
 #### --distractors--
 
-Placeholder distractor 1
+Primary Key
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:244]:
```suggestion
Primary Key.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 12]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 2
+Unique Key
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:248]:
```suggestion
Unique Key.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 13]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 3
+Composite Key
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:252]:
```suggestion
Composite Key.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 14]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --answer--
 
-Placeholder answer
+Foreign Key
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:256]:
```suggestion
Foreign Key.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 15]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --text--
 
-Placeholder question
+What does Third Normal Form (3NF) eliminate?
 
 #### --distractors--
 
-Placeholder distractor 1
+Multi-Valued Dependencies
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:266]:
```suggestion
Multi-Valued Dependencies.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 16]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 2
+Partial Dependencies
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:270]:
```suggestion
Partial Dependencies.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 17]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 3
+All Dependencies
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:274]:
```suggestion
All Dependencies.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 18]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --answer--
 
-Placeholder answer
+Transitive Dependencies
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:278]:
```suggestion
Transitive Dependencies.
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 19]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`WHERE`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What happens to child records when implementing "ON DELETE CASCADE"?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:306]:
```suggestion
What happens to child records when implementing `ON DELETE CASCADE`?
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 20]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+They are automatically deleted.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the "COALESCE" function used for?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:328]:
```suggestion
What is the `COALESCE` function used for?
```

File: curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md [hunk 21]
```diff
@@ -17,439 +17,439 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+Partial Dependency.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What's the main risk of using "SELECT *" in production code?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md:372]:
```suggestion
What's the main risk of using `SELECT *` in production code?
```

**Review by moT01 — CHANGES_REQUESTED** [review]
Pretty minor, but some more small improvements - or maybe just my preference in the case of adding the periods

**Review by moT01 — APPROVED** [review]

**Review by Sembauke — APPROVED** [review]

**Pull request merged**